pkg/controllers/policycache/controller.go

```diff
@@ -72,8 +70,7 @@ func (c *controller) WarmUp() error {
 		if key, err := cache.MetaNamespaceKeyFunc(policy); err != nil {
 			return err
 		} else {
-			subresourceGVKToKind := getSubresourceGVKToKindMap(policy, c.client)
-			c.cache.Set(key, policy, subresourceGVKToKind)
+			return c.cache.Set(key, policy, c.client.Discovery())
```

Comment:
I'm not sure adding the dependency on the client is a good idea. Though the policy cache is a small portion in the webhook controller for fast lookup and it is not used in any other background controllers, making it independent could be easier for us to re-use it (i.e. the cleanup controller).

Reply:
How can we do without this ?

Reply:
Discussed with @eddycharly over a call, we need to check how this is used in the CLI and if we can support it with the client. I think it's ok to get this PR merged now, @eddycharly will get back with follow-up PRs if we need more changes for the CLI.